server/controllers/patient.js

```diff
@@ -96,10 +96,15 @@ exports.searchFuzzy = function (req, res, next) {
     'WHERE LEFT(LOWER(CONCAT(p.last_name, \' \', p.first_name )), CHAR_LENGTH(?)) = ? OR ' +
       'SOUNDEX(p.first_name) = SOUNDEX(?) OR ' +
       'SOUNDEX(p.last_name) = SOUNDEX(?) OR ' +
-      'SOUNDEX(CONCAT(p.last_name, \' \', p.first_name)) = SOUNDEX(?) ' +
+      'SOUNDEX(p.middle_name) = SOUNDEX(?) OR ' +
+      'SOUNDEX(CONCAT(p.last_name, \' \', p.first_name)) = SOUNDEX(?) OR ' +
+      'SOUNDEX(CONCAT(p.first_name, \' \', p.last_name)) = SOUNDEX(?) OR ' +
+      'SOUNDEX(CONCAT(p.first_name, \' \', p.last_name, \' \', p.middle_name)) = SOUNDEX(?) OR ' +
+      'SOUNDEX(CONCAT(p.last_name, \' \', p.middle_name, \' \', p.first_name)) = SOUNDEX(?) OR ' +
```

Comment:
The way SOUNDEX works, you don't need to have lines 102 and 103.  They will product e the same as lines 100 and 101 for shorter inputs.  I think this is doing too much SQL work, and may take a long time when we have many patients...